samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java

```diff
@@ -125,7 +126,12 @@ Config generateConfig() {
     if (isIntermediate()) {
       streamConfig.put(String.format(StreamConfig.IS_INTERMEDIATE_FOR_STREAM_ID(), spec.getId()), "true");
       streamConfig.put(String.format(StreamConfig.DELETE_COMMITTED_MESSAGES_FOR_STREAM_ID(), spec.getId()), "true");
-      streamConfig.put(String.format(StreamConfig.CONSUMER_OFFSET_DEFAULT_FOR_STREAM_ID(), spec.getId()), "oldest");
+
+      // Setting offset.default to oldest only if the job is running in batch mode
+      if (new ApplicationConfig(config).getAppMode().equals(ApplicationConfig.ApplicationMode.BATCH)) {
```

Comment:
nit: I'd suggest to do this other way round to guard against getAppMode() returning potential nulls even though currently it seems to return STREAM by default.
`ApplicationConfig.ApplicationMode.BATCH.equals(...)`